### ISSUE TITLE
cm ssh

### ISSUE THREAD

**Issue by laszewsk** [cloudmesh/cloudmesh-cloud#224]
a) cms vm ssh should do the same as cm ssh

b) cm ssh returns an error after 

cms init
cms vm boot
cms ssh

cloudmesh-cloud/cloudmesh/ssh/command/ssh.py", line 173, in do_ssh
    vm = cm.find_name(names[0], kind="vm")[0]
IndexError: list index out of range

**Comment by Vafa-Andalibi** [comment]
`cms ssh`  is not part of the `ssh` command docopt. Did you omit the rest of the command here in this issue or you used `cms ssh` instead of something like `cms ssh --name=vafandal-container-54 whoami` ? this command  `cms ssh --name=vafandal-container-54 whoami`  works fine for me

**Comment by laszewsk** [comment]
Hmmm ... look slike its committed

cms help ssh
Command ssh
===========

::

    Usage:
        ssh config list [--output=OUTPUT]
        ssh config add NAME IP [USER] [KEY]
        ssh config delete NAME
        ssh host delete NAME
        ssh host add NAME
        ssh [--name=VMs] [--user=USERs] [COMMAND]

    Arguments:
      NAME        Name or ip of the machine to log in
      list        Lists the machines that are registered and
                  the commands to login to them
      PARAMETERS  Register te resource and add the given
                  parameters to the ssh config file.  if the
                  resource exists, it will be overwritten. The
                  information will be written in /.ssh/config

    Options:
       -v                verbose mode
       --output=OUTPUT   the format in which this list is given
                         formats includes cat, table, json, yaml,
                         dict. If cat is used, it is just printed as
                         is. [default: table]
       --user=USERs      overwrites the username that is
                         specified in ~/.ssh/config
       --name=CMs        the names of the VMS to execute the
                         command on

    Description:
        ssh config list
            lists the hostsnames that are present in the
            ~/.ssh/config file

        ssh config add NAME IP [USER] [KEY]
            registers a host i ~/.ssh/config file
            Parameters are attribute=value pairs
            Note: Note yet implemented

        ssh [--name=VMs] [--user=USERs] [COMMAND]
            executes the command on the named hosts. If user is
            specified and is greater than 1, it must be specified for
            each vm. If only one username is specified it is used for
            all vms. However, as the user is typically specified in the
            cloudmesh database, you probably do not have to specify
            it as it is automatically found.

    Examples:


         ssh config add blue 192.168.1.245 blue

             Adds the following to the !/.ssh/config file

             Host blue
                  HostName 192.168.1.245
                  User blue
                  IdentityFile ~/.ssh/id_rsa.pub

**Comment by Vafa-Andalibi** [comment]
are we supposed to have a `cms ssh` command ? without any extra arguments?

**Comment by laszewsk** [comment]
yes, i found for example that vagrant has a similar logic e.g. we log into the last vm we know about 

cms vm ssh  equals cms ssh we just have two things for it. however cms vm ssh, doesn not work on ssh config. it only starts the vm

**Comment by Vafa-Andalibi** [comment]
how do we define and determine the "last vm" ?

**Comment by laszewsk** [comment]
When you look into the code for cm boot, it writes the last name in 

Variables()

You can get it from 

variables = Variables().name

Than we assume you use 

names = Parameter.expand(variables)

Than the last name is supposed to be names[0]

In principal we only have to store the last vm

I have not thought about if we should just add to names and have that more or less an array

E.g. vm0,vm1,….

**Comment by Vafa-Andalibi** [comment]
are we supposed to get a shell? or just run command like `ssh COMMAND` ?  
I can't find a command that actually ssh to a vm and get a interactive shell for the user through cms. Can you give me an example? `cms vm ssh`  is just trying to run a command

**Comment by laszewsk** [comment]
If you specify the command the command is run, if you do not dod a command we log into the shell.

We can do this as we use our public key in ~/.ssh and push this out before we start the vm to the cloud

**Comment by Vafa-Andalibi** [comment]
fixed with commit db986db3e693ec8e9ce36251cc5b7ab40c3b9b36
